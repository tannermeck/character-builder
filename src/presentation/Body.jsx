import React from 'react';

export default function Body({ head, onHeadChange, torso, onTorsoChange, legs, onLegChange }){
    const headOptions = ['bird', 'cat', 'dog', 'goat', 'pig']
    const torsoOptions = ['black-blouse', 'blue-shirt', 'red-blouse', 'sweater', 'tuxedo']
    const legOptions = ['blue-shorts', 'grey-jeans', 'speedo', 'swimtrunks', 'elf']
    return (
    <>
        <section>
            <h1>Head:</h1>
            <div>
                <select value={head} onChange={(e) => onHeadChange(e.target.value)}>
                    {headOptions.map(option => {
                        return <option key={option}>{option}</option>
                    })}
                </select>
            </div>
        </section>
        <section>
            <h1>Torso:</h1>
            <div>
                <select value={torso} onChange={(e) => onTorsoChange(e.target.value)}>
                    {torsoOptions.map(option => {
                        return <option key={option}>{option}</option>
                    })}
                </select>
            </div>
        </section>
        <section>
            <h1>Legs:</h1>
            <div>
                <select value={legs} onChange={(e) => onLegChange(e.target.value)}>
                    {legOptions.map(option => {
                        return <option key={option}>{option}</option>
                    })}
                </select>
            </div>
        </section>
    </>
    )
}