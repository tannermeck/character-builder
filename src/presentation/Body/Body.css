.catchphraseContainer {
    width: 170px;
    display: flex;
    flex-direction: column;
    justify-content: left;
    padding-top: 50px;
}
.catchphraseLabel {
    font-weight: bolder;
    padding-bottom: 10px;
}
.catchphraseContainer button {
    width:fit-content;
    max-width: 100%;
    margin-top: 10px;
}