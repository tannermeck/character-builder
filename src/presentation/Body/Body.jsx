import React from 'react';
import './Body.css'

export default function Body({ head, onHeadChange, torso, onTorsoChange, legs, onLegChange, handleClick, setCatchphrase }){
    const headOptions = ['bird', 'cat', 'dog', 'goat', 'pig']
    const torsoOptions = ['black-blouse', 'blue-shirt', 'red-blouse', 'sweater', 'tuxedo']
    const legOptions = ['blue-shorts', 'grey-jeans', 'speedo', 'swimtrunks', 'elf']
    return (
    <>
        <section>
            <h1>Head:</h1>
            <div>
                <select value={head} onChange={(e) => onHeadChange(e.target.value)}>
                    {headOptions.map(option => {
                        return <option key={option}>{option}</option>
                    })}
                </select>
            </div>
        </section>
        <section>
            <h1>Torso:</h1>
            <div>
                <select value={torso} onChange={(e) => onTorsoChange(e.target.value)}>
                    {torsoOptions.map(option => {
                        return <option key={option}>{option}</option>
                    })}
                </select>
            </div>
        </section>
        <section>
            <h1>Legs:</h1>
            <div>
                <select value={legs} onChange={(e) => onLegChange(e.target.value)}>
                    {legOptions.map(option => {
                        return <option key={option}>{option}</option>
                    })}
                </select>
            </div>
        </section>
        <div className="catchphraseDivContainer">
            <label className="catchphraseLabel">Add a Catchphrase:</label>
            <input type="text" onChange={(e) => setCatchphrase(e.target.value)}/>
            <button onClick={(e) => handleClick(e.target.value)}>Add Catchphrase</button>
        </div>
    </>
    )
}