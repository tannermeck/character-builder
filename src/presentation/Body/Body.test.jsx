import { render } from '@testing-library/react';
import Body from './Body';

it('should render a snapshot test of the body', () => {
    const { container } = render(<Body />)
    expect(container).toMatchSnapshot();
})