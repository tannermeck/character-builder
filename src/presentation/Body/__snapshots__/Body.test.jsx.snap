// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a snapshot test of the body 1`] = `
<div>
  <section>
    <h1>
      Head:
    </h1>
    <div>
      <select>
        <option>
          bird
        </option>
        <option>
          cat
        </option>
        <option>
          dog
        </option>
        <option>
          goat
        </option>
        <option>
          pig
        </option>
      </select>
    </div>
  </section>
  <section>
    <h1>
      Torso:
    </h1>
    <div>
      <select>
        <option>
          black-blouse
        </option>
        <option>
          blue-shirt
        </option>
        <option>
          red-blouse
        </option>
        <option>
          sweater
        </option>
        <option>
          tuxedo
        </option>
      </select>
    </div>
  </section>
  <section>
    <h1>
      Legs:
    </h1>
    <div>
      <select>
        <option>
          blue-shorts
        </option>
        <option>
          grey-jeans
        </option>
        <option>
          speedo
        </option>
        <option>
          swimtrunks
        </option>
        <option>
          elf
        </option>
      </select>
    </div>
  </section>
  <div
    class="catchphraseDivContainer"
  >
    <label
      class="catchphraseLabel"
    >
      Add a Catchphrase:
    </label>
    <input
      type="text"
    />
    <button>
      Add Catchphrase
    </button>
  </div>
</div>
`;
